src/logic/currencyValues/api/fetchCurrenciesRates.ts

```diff
@@ -1,35 +1,39 @@
 import axios from 'axios'
 
-import { getExchangeRatesUrl, getExchangeRatesUrlFallback } from 'src/config'
+import { getExchangeRatesUrl } from 'src/config'
 import { AVAILABLE_CURRENCIES } from '../store/model/currencyValues'
+import fetchTokenCurrenciesBalances from './fetchTokenCurrenciesBalances'
+import BigNumber from 'bignumber.js'
 
 const fetchCurrenciesRates = async (
   baseCurrency: AVAILABLE_CURRENCIES,
   targetCurrencyValue: AVAILABLE_CURRENCIES,
+  safeAddress: string,
 ): Promise<number> => {
   let rate = 0
-  const url = `${getExchangeRatesUrl()}?base=${baseCurrency}&symbols=${targetCurrencyValue}`
-  const fallbackUrl = `${getExchangeRatesUrlFallback()}?currency=${baseCurrency}`
+
+  if (targetCurrencyValue === AVAILABLE_CURRENCIES.ETH) {
+    try {
+      const result = await fetchTokenCurrenciesBalances(safeAddress)
+      if (result?.data && result?.data.length > 0) {
+        rate = new BigNumber(1).div(new BigNumber(result.data[0].usdConversion)).toNumber()
```

Comment:
I'm not sure about this, but I'm almost certain that you don't need to instantiate BigNumber for the `div` param

Something around:
```suggestion
        rate = new BigNumber(1).div(result.data[0].usdConversion).toNumber()
```

Reply:
Yes me neither, wasn't sure, but If you think it's ok lets do it :)